tktermwidget/style.py

```diff
@@ -1,27 +1,275 @@
 """Styles for terminal"""
+from __future__ import annotations
 
-Default: dict = {  # Style for normal tkterminalwidget
+from json import dump, load
+from pathlib import Path
+
+from platformdirs import user_cache_dir
+
+# Constants
+STYLE_PATH = Path(user_cache_dir("tktermwidget"))
+JSON_FILE = STYLE_PATH / "styles.json"
+
+# Check the style file
+if not STYLE_PATH.exists():
+    STYLE_PATH.mkdir(parents=True)
+    with open(JSON_FILE, "w", encoding="utf-8") as f:
+        dump("{}", f)
+
+if not (JSON_FILE).exists():
+    with open(JSON_FILE, "w", encoding="utf-8") as f:
+        dump("{}", f)
+
+
+# Create style
+def write_style(**styles) -> None:
+    """Write the style into the json file"""
+    # User can use this function to write with their own style
+    # TODO: improve it, new style will overwrite the old style
+    with open(JSON_FILE, "w", encoding="utf-8") as json_obj:
+        dump(styles, json_obj, indent=1)
+
+
+# Load style
+def load_style() -> dict:
+    """Get style from the json file"""
+    # User also can use this function to get their own style
+    with open(JSON_FILE, "r", encoding="utf-8") as json_obj:
+        return load(json_obj)
+
+
+# Config style
+def config_style() -> None:
```

Comment:
Make this into an app class instead to remove nested functions

Reply:
Okay, I will do it.

Reply:
I rebuilded it and wrote a lot
see ```rebuild style.py``` commit